tests/js/server/aql/aql-queries-simple.js

```diff
@@ -1335,6 +1335,18 @@ function ahuacatlQuerySimpleTestSuite () {
       });
     },
 
+    testLargeQuery : function() {
+      let q = "";
+      const cnt = 1000;
+      for (let i = 0; i < cnt; ++i) {
+        q += `LET v${i} = NOOPT(1)\n`;
+      }
+      q += "RETURN v0";
+      for (let i = 1; i < cnt; ++i) {
+        q += ` + v${1}`;
+      }
+      assertEqual([cnt], AQL_EXECUTE(q, {}, {optimizer: {rules: ['-all']}}).json);
+    }
```

Comment:
I would like to ask for a second Test like the one you already have,
but with the CallStackOverflow hidden within a Subquery (see my comment in the Optimizer Rule)

Reply:
Done.